### PR TITLE
feat: add allow-overwrite input

### PR DESCRIPTION
**Pull request by chrismanley**

### CODE REVIEW

**Review by egilsster — CHANGES_REQUESTED** [review]
We added checks to fail when files existed because overwriting the files burned us in the past. What's your use case? Can you elaborate a bit on this

**Comment by chrismanley** [comment]
It burned me in particular, so I'm well familiar! 

The use case is for our master branch. We want to have it always building and I don't want to create a unique deploy for it on every single commit. Disallowing overwrite is still the default, this would only be used when explicitly set.

File: qmfe-to-aws/qmfe-to-aws.sh [hunk 1]
```diff
@@ -75,7 +79,7 @@ aws --version
 aws configure set aws_access_key_id "$AWS_ACCESS_KEY_ID" --profile qcs-cdn
 aws configure set aws_secret_access_key "$AWS_SECRET_ACCESS_KEY" --profile qcs-cdn
 
-if [[ $(aws s3 ls "s3://$AWS_BUCKET_NAME/$S3_KEY/$QMFE_ID/$VERSION" | head) ]]; then
+if [[ $ALLOW_OVERWRITE = "false" ]] && [[ $(aws s3 ls "s3://$AWS_BUCKET_NAME/$S3_KEY/$QMFE_ID/$VERSION" | head) ]]; then
```
Comment (nilzona) [inline qmfe-to-aws/qmfe-to-aws.sh:82]: shell script is not like any other language. I think you should use `==` here according to the docs I found here https://linuxize.com/post/how-to-compare-strings-in-bash/

File: qmfe-to-aws/action.yaml [hunk 2]
```diff
@@ -47,6 +47,10 @@ inputs:
     description: "Cache control headers"
     required: false
     default: "public,max-age=31536000,s-maxage=2629800,immutable"
+  allow-overwrite:
```
Comment (nilzona) [inline qmfe-to-aws/action.yaml:50]: perhaps a disclaimer here *WARNING*  or something

**Review by nilzona — COMMENTED** [review]
Since this would be a conscious decision to actually allow overwrite I think it's ok for the use case you provided. Normally users will not see this. But maybe make it a bit clearer in the docs that it should just be used if you "know what youi're doing"

**Review by egilsster — APPROVED** [review]

**Pull request merged**